frappe/auth.py

```diff
@@ -209,6 +215,23 @@ def authenticate(self, user=None, pwd=None):
 		self.check_if_enabled(user)
 		self.user = self.check_password(user, pwd)
 
+	def force_user_to_reset_password(self):
+		if not self.user:
+			return
+
+		reset_pwd_after_days = cint(frappe.db.get_single_value("System Settings",
+			"force_user_to_reset_password"))
+
+		if reset_pwd_after_days:
+			last_password_reset_date = frappe.get_cached_value("User",
```

Comment:
cached value not required

Reply:
if date is not set, set it right now instead of forcing user to reset

Reply:
If date is not set then it will not force user to reset the password, because date diff will return the value as zero.